File: Java version/Errors/F0FErrors.java
package Errors;

import Tokens.*;
import Tokens.F0FToken.TokenType;

public class F0FErrors {
    public static void error(int line, String message) {
        report(line, "", message);
    }

    private static void report(int line, String where, String message) {
        System.err.println("[line " + line + "] Error" + where + ": " + message);
        //hadError = true;
    }

    public static void error(F0FToken token, String message) {
        if (token.type == TokenType.EOF) {
            report(token.line, " at end", message);
        } 
        else {
            report(token.line, " at '" + token.lex + "'", message);
        }
    }
}

File: F0F/src/F0FErrors.py
from F0FTokens import Token, TokenType, TerminalsTokens, NonTerminalsTokens, symbols_tokens, keywords_tokens

class F0FError:
    def __init__(self,line:int, msg:str):
        self.message = msg
        self.line = line
    
    def __str__(self) -> str:        
        rep = "[line " + str(self.line) + "] : " + self.message
        return rep
    def __repr__(self):
        return self.__str__()
class LexerError(F0FError):
    pass
class ParsingError(F0FError):
    """
    Base class for all parsing exceptions.
    """
    def __init__(self, token:Token, msg:str):
        super().__init__(token.line,msg)
        self._token_prblm = token
    def token_with_problem(self):
        return self._token_prblm
    def __str__(self) -> str:        
        rep = "Parsing error at: " + str(self._token_prblm.lex) +" [line " + str(self.line) + "] : " + self.message
        return rep